app/views/self_service/onboarding/_firms_index.html.erb

```diff
@@ -0,0 +1,14 @@
+<% if principal.next_onboarding_action == :complete_a_firm %>
+  <div class="onboarding-message callout">
+    <%= paragraphs t('self_service.onboarding.firms_index.complete_a_firm_callout') %>
+  </div>
+<% end %>
+<% if principal.next_onboarding_action == :complete_an_adviser %>
+  <div class="onboarding-message callout">
+    <%= paragraphs t('self_service.onboarding.firms_index.complete_an_adviser_callout') %>
+    <p>
+      <% firm = principal.main_firm_with_trading_names.registered.first %>
+      <%= link_to t('self_service.onboarding.firms_index.complete_an_adviser_button', firm_name: firm.registered_name), new_self_service_firm_adviser_path(firm), class: 'button button--primary' %>
+    </p>
+  </div>
+<% end %>
```

Comment:
Seems like there's a lot of duplication in these. Is it not possible to extract the commonality out and pass in things like 'firm_index' or 'firm_form' as locals?